forge-game/src/main/java/forge/game/trigger/TriggerDamageDoneOnce.java

```diff
@@ -59,6 +59,10 @@ public void setTriggeringObjects(SpellAbility sa, Map<AbilityKey, Object> runPar
         }
         sa.setTriggeringObject(AbilityKey.Target, target);
         sa.setTriggeringObject(AbilityKey.Sources, getDamageSources(damageMap));
+        for (final Map.Entry<Card, Integer> entry : damageMap.entrySet()) {
```

Comment:
add a TODO for Two-Headed-Giant

otherwise i need to look for a per Controller Trigger variant ...

Ah i found a corner case:
* If you have the Initiative, and opponent attacks you with 2 creatures.
* One of them is unblocked, the other is blocked.
* But the Blocked creature has a damage redirect to You.
* So what happens is that the Unblocked Creature would make Opponent gain the Initiative
* and also your blocking creature dealt damage to you so you also would gain the Initiative
* This should be two Triggers you can stack like you want.

Reply:
i think i will create a new trigger for this

i will ask Judges about this

Reply:
Thanks :)